File: AdminArea/AdminArea.Web/Modules/souqcom/Cartcom/CartcomForm.cs
using Serenity.ComponentModel;

namespace AdminArea.souqcom.Forms;

[FormScript("souqcom.Cartcom")]
[BasedOnRow(typeof(CartcomRow), CheckNames = true)]
public class CartcomForm
{
    public int UserId { get; set; }
    public int ProductId { get; set; }
    public int Qty { get; set; }
}

File: AdminArea/AdminArea.Web/Modules/ServerTypes/souqcom/CartcomForm.ts
import { IntegerEditor, ServiceLookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CartcomForm {
    UserId: IntegerEditor;
    ProductId: ServiceLookupEditor;
    Qty: IntegerEditor;
}

export class CartcomForm extends PrefixedContext {
    static readonly formKey = 'souqcom.Cartcom';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CartcomForm.init)  {
            CartcomForm.init = true;

            var w0 = IntegerEditor;
            var w1 = ServiceLookupEditor;

            initFormType(CartcomForm, [
                'UserId', w0,
                'ProductId', w1,
                'Qty', w0
            ]);
        }
    }
}